pallets/prices/src/lib.rs

```diff
@@ -124,14 +142,30 @@ impl<T: Config> PriceFeeder for Pallet<T> {
     fn get_price(currency_id: &CurrencyId) -> Option<PriceDetail> {
         // if emergency price exists, return it, otherwise return latest price from oracle.
         Self::get_emergency_price(currency_id).or_else(|| {
-            T::Source::get(&currency_id).and_then(|p| {
-                10u128.checked_pow(p.value.decimal.into()).and_then(|d| {
-                    p.value
-                        .price
-                        .checked_div(&FixedU128::from_inner(d))
-                        .map(|price| (price, p.timestamp))
+            if currency_id == &T::LiquidCurrency::get() {
+                T::Source::get(&T::StakingCurrency::get()).and_then(|p| {
+                    10u128.checked_pow(p.value.decimal.into()).and_then(|d| {
+                        p.value
+                            .price
+                            .checked_div(&FixedU128::from_inner(d))
```

Comment:
You already have decimal in prices info, why still use from_inner here?

Reply:
this is the only place that we care about decimal on-chain, while other pallet query prices from this `price pallet`, will return a minimum unit of currency price, so we need to divide the decimal.